core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -157,6 +160,44 @@ abstract class AppBase(
   var mlEventLogType = ""
   var pysparkLogFlag = false
 
+  def getOrCreateExecutor(executorId: String, addTime: Long): ExecutorInfoClass = {
+    executorIdToInfo.getOrElseUpdate(executorId, {
+      new ExecutorInfoClass(executorId, addTime)
+    })
+  }
+
+  /**
+   * Retrieves cluster information based on executor nodes.
+   * If executor nodes exist, calculates the number of hosts and total cores,
+   * and extracts executor and driver instance types (databricks only)
+   *
+   * @return Cluster information including cores, number of nodes, and instance types.
+   */
+  def getClusterInfo: Option[ClusterInfo] = {
+    // TODO: Handle dynamic allocation when determining the number of nodes.
+    sparkProperties.get("spark.dynamicAllocation.enabled").foreach { value =>
+      if (value.toBoolean) {
+        logWarning("Dynamic allocation is not supported. Cluster information may be inaccurate.")
+      }
+    }
+    val activeExecInfo = executorIdToInfo.values.collect {
+      case execInfo if execInfo.isActive => (execInfo.host, execInfo.totalCores)
+    }
+    if (activeExecInfo.nonEmpty) {
+      val (activeHosts, coresPerExecutor) = activeExecInfo.unzip
+      if (coresPerExecutor.toSet.size != 1) {
+        logWarning("Cluster with variable executor cores detected. Using maximum value.")
```

Comment:
same as the above comment for the identity of the app in the log message.

Reply:
Added `appId` in logs